text/3-entrees/randomness.md

```diff
@@ -0,0 +1,74 @@
+
+These resources may also be helpful in assessing the security and limitations of these randomness sources.
+
+* https://github.com/paritytech/ink/issues/57
+* https://wiki.polkadot.network/docs/en/learn-randomness
+
+## Randomness Trait
+
+The randomness trait provides two methods, `random_seed`, and `random`, both of which provide a pesudo-random value of the type specified in the traits type parameter.
+
+### `random_seed`
+
+The `random_seed` method takes no parameters and returns a random seed which changes once per block. If you call this method twice in the same block you will get the same result. This method is typically not as useful as its counterpart.
+
+### `random`
+
+The `random` method takes a byte array, `&[u8]`, known as the subject, and uses the subject's bytes along with the random seed described in the previous section to calculate a final random value. Using a subject in this way allows pallet (or multiple pallets) to seek randomness in the same block and get different results. The subject does not add entropy or security to the generation process, it merely prevents each call from returning identical values.
+
+Common values to use for a subject include:
+* The block number
+* The caller's accountId
+* A Nonce
+* A pallet-specific identifier
+* A tuple containing several of the above
+
+To bring a randomness source into scope, we include it in our configuration trait with the appropriate trait bound. This pallet, being a demo, will use two different sources. Using multiple sources is not necessary in practice.
+
+```rust, ignore
+pub trait Trait: system::Trait {
+	type Event: From<Event> + Into<<Self as system::Trait>::Event>;
+
+	type CollectiveFlipRandomnessSource: Randomness<H256>;
+
+	type BabeRandomnessSource: Randomness<H256>;
+}
+```
+
+We've provided the `Output` type as [`H256`](https://substrate.dev/rustdocs/master/sp_core/struct.H256.html).
+
+## Collective Coin Flipping
+
+Substrate's [Randomness Collective Flip pallet](https://substrate.dev/rustdocs/master/pallet_randomness_collective_flip/index.html) uses a safe mixing algorithm to generate randomness using the entropy of previous block hashes. Because it is dependent on previous blocks, it can take many blocks for the seed to change.
+
+A naive randomness source based on block hashes would take the hash of the previous block and use it as a random seed. Such a technique has the significant disadvantage that the block author can preview the random seed, and choose to discard the block choosing a slightly modified block with a more desirable hash.
+
+This pallet reduces the block author's influence over the random seed, by distributing that influence to the authors of the last 81 blocks.
```

Comment:
Remove sentence.  Avoid indicating that Randomness Collective Flip ever provides any security properties

Reply:
As for the previous paragraph, not sure.  I'd just say this only exists for testing and should never be used in production.